plugins/services/src/js/containers/service-detail/ServiceDetail.js

```diff
@@ -181,60 +153,54 @@ class ServiceDetail extends mixin(TabsMixin) {
 
   getTabs() {
     const {service:{id}} = this.props;
-    const routePrefix = `/services/overview/${encodeURIComponent(id)}`;
+    const routePrefix = `/services/detail/${encodeURIComponent(id)}`;
 
-    const tabs = [];
-    const activeTab = this.state.currentTab;
-
-    tabs.push({
-      label: 'Instances',
-      callback: () => {
-        this.setState({currentTab: 'tasks'});
-      },
-      isActive: activeTab === 'tasks'
-    });
-
-    tabs.push({
-      label: 'Configuration',
-      callback: () => {
-        this.setState({currentTab: 'configuration'});
-      },
-      isActive: activeTab === 'configuration'
-    });
-
-    tabs.push({
-      label: 'Debug',
-      callback: () => {
-        this.setState({currentTab: 'debug'});
-      },
-      isActive: activeTab === 'debug'
-    });
+    const tabs = [
+      {label: 'Instances', routePath: `${routePrefix}/tasks`},
+      {label: 'Configuration', routePath: `${routePrefix}/configuration`},
+      {label: 'Debug', routePath: `${routePrefix}/debug`}
+    ];
 
     if (this.hasVolumes()) {
       tabs.push({
-        label: 'Volumes', routePath: routePrefix + '/volumes',
-        callback: () => {
-          this.setState({currentTab: 'volumes'});
-        },
-        isActive: activeTab === 'volumes'
+        label: 'Volumes',
+        routePath: `${routePrefix}/volumes`
       });
     }
 
     return tabs;
   }
 
   render() {
-    const {children, service:{id}} = this.props;
-    const breadcrumbs = <ServiceBreadcrumbs serviceID={id} />;
+    const {children, actions, errors, params, routes, service} = this.props;
+    const breadcrumbs = <ServiceBreadcrumbs serviceID={service.id} />;
+    const clonedProps = {
+      params,
+      routes,
+      service,
+      errors: MarathonErrorUtil.parseErrors(errors[ActionKeys.SERVICE_EDIT]),
+      onClearError: this.handleEditClearError,
+      onEditClick: actions.editService,
+      volumes: service.getVolumes().getItems()
+    };
+
+    const clonedChildren = React.Children.map(children, function (child) {
+      // Only add props to children that are not ServiceModals
+      if (child.type === ServiceModals) {
+        return child;
+      }
+
+      return React.cloneElement(child, clonedProps);
```

Comment:
Because we need to pass down extra props to the elements. Props that was being passed down in these render functions that we turned into routes: https://github.com/dcos/dcos-ui/pull/2058/files#diff-fc8d30a093163d01bb787f96fdf40542L94